packages/@aws-cdk/aws-stepfunctions/lib/fields.ts

```diff
@@ -214,8 +214,10 @@ export class FieldUtils {
 }
 
 function validateJsonPath(path: string) {
-  if (path !== '$' && !path.startsWith('$.') && path !== '$$' && !path.startsWith('$$.')) {
-    throw new Error(`JSON path values must be exactly '$', '$$', start with '$.' or start with '$$.' Received: ${path}`);
+  if (path !== '$' && !path.startsWith('$.')
+    && path !== '$$' && !path.startsWith('$$.')
+    && !/^\$\[\d+\]$/.test(path) && !/^\$\[\d+\]\./.test(path)) {
```

Comment:
maybe put one condition per line to make this more readable.
should this just test for begins with `[` rather than the check for digits

Json path specification supports bracket or dot notation.
i.e. `$.field` is the same as `$['field']`